.github/workflows/lint.yml

```diff
@@ -0,0 +1,62 @@
+name: Linting
+
+on:
+  pull_request:
+  push:
+    branches: [main]
+  workflow_dispatch:
+
+defaults:
+  run:
+    shell: bash
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.ref }}
+  cancel-in-progress: true
```

Comment:
what does this do?

Reply:
The concurrency is the same as for the other workflow.
This ensures that pushing a second commit on the same branch kills the workflow running for the previous commit, to avoid wasting computational resources.
This is not necessary for lining as this is super quick but I kept it to be uniform.

Reply:
makes sense, thanks!